source/agora/node/FullNode.d

```diff
@@ -1026,11 +1026,36 @@ public class FullNode : API
     {
         this.recordReq("postEnrollment");
 
+        /// FIXME: Use a proper type and sensible memory allocation pattern
+        version (all)
+        {
+            scope extraSet = new TestUTXOSet();
+            foreach (ref Hash hash, ref Transaction tx; this.pool)
+                extraSet.put(tx);
+            scope extraFinder = extraSet.getUTXOFinder();
+            scope UTXOFinder enrollmentsUTXOFinder =
+                (in Hash utxo, out UTXO val)
+                {
+                    if (this.utxo_set.peekUTXO(utxo, val))
```

Comment:
Why is this using `peekUTXO` ? It just bypasses any replay protection.

Reply:
Right. We should use `UTXOFinder`.